src/box/index_def.c

```diff
@@ -224,13 +310,18 @@ index_opts_dup(const struct index_opts *opts, struct index_opts *dup)
 {
 	*dup = *opts;
 	if (dup->covered_fields != NULL) {
-		uint32_t *fields = dup->covered_fields;
+		struct covered_field *fields = dup->covered_fields;
 		dup->covered_fields =
 			xmalloc(dup->covered_field_count *
 				sizeof(*dup->covered_fields));
-		memcpy(dup->covered_fields, fields,
-		       dup->covered_field_count *
-		       sizeof(*dup->covered_fields));
+		for (uint32_t i = 0; i < dup->covered_field_count; i++) {
+			dup->covered_fields[i].field = fields[i].field;
+			if (fields[i].layout == NULL)
+				dup->covered_fields[i].layout = NULL;
+			else
+				dup->covered_fields[i].layout =
+					xstrdup(fields[i].layout);
+		}
```

Comment:
We need to patch `index_opts_destroy()` too as now `layout` field need to be freed. The ASAN would catch this probably.